docs/pages/web/alertmodal.js

```diff
@@ -0,0 +1,235 @@
+            description="Use to overlay Page content; horizontally and vertically center modals on the screen."
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Use AlertModal on top of another Modal Dialog. This can cause accessibility issues with focus states and make it hard for a user to escape and go back to the previous surface. On mobile surfaces, if you have to confirm something triggered by a modal dialog, auto-dismiss the first dialog before presenting with the confirmation."
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            description="Limit the content to prevent the need to scroll at most screen sizes."
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Use AlertModal for content for long and complex tasks or that should have a dedicated surface, like login flows. If you have to put more functionality in an overlay, use GenericModal, or Sheet."
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            description="Provide a way for the user to correct an error or issue via a button or a link."
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Leave it up to the user to find where to go to fix an issue."
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            description="Explain to the user why they’ve encountered a warning or error when an action button or link is not possible."
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Omit an explanation as to why a user is encountering an error or issue."
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <AccessibilitySection name={generatedDocGen?.displayName}>
+        <MainSection.Subsection
+          title="Labels"
+          description={`
+          We want to make sure AlertModals have a clear purpose when being read by a screen reader. \`accessibilityModalLabel\` allows us to update the spoken text for the heading prop and give it more context.`}
+        >
+          <MainSection.Card cardSize="lg" />
+        </MainSection.Subsection>
+      </AccessibilitySection>
+
+      <MainSection
+        name="Localization"
+        description={`Be sure to localize the \`heading\` and \`accessibilityModalLabel\`props, as well as any other text elements within AlertModal. Note that localization can lengthen text by 20 to 30 percent. `}
+      />
+      <MainSection name="Variants">
```

Comment:
I'm concerned that there's nothing in the code to enforce these variants, which don't seem to correspond to `type`? If we're recommending specific versions of this component, we should make those easy to build, and make it hard/impossible to build versions that don't match those variants/guidelines. I could currently build something like
```js
<AlertModal
  type="default"
  heading="Nope, that's wrong"
  primaryAction={{
    accessibilityLabel: 'Oops',
    label: "C'est la vie",
  }}
  onDismiss={handleDismiss}
>
Whatever you just did isn't correct. Are you sure you want to continue?
</AlertModal>
```
…which really doesn't seem like something we'd ever want someone to build. The wrong `type` for the message, no secondary "cancel" action required, an action that doesn't actually do anything (neither `href` nor `onClick` are required). Are we ok with this? If we have specific guidelines for variants, should we guide users to those variants via the component API?

Reply:
The guidelines around actions aren't strict enough that we could enforce through props, imo. Errors and warnings sometimes have one action, sometimes have two. I think the only thing we could safeguard for is ensuring either onClick or href exists

Reply:
Ok, the href/onClick check helps. There's still the question of if we want to offer a limited number of specific types of AlertModals, or if the user is free to build whatever they need (given the already constrained props available).

Assuming it's the latter, we may need to reconsider the docs structure here. Most other pages use variants to explain specific props, but these seem more like guidelines or patterns. I took the variants to mean "these are specific types we offer", and I wonder if others would be confused as well. What do you think?

Reply:
We could rename to be Error, Warning, Multiple Actions, Single Action to be more prop-based, or like Type - Error, Type- Warning, For acknowledgments, and For confirmations?

Reply:
And yeah, in this case it's the latter

Reply:
Maybe both? I think the guidance you're currently providing is really valuable, I just want it to be clear that it's _guidance_ and not something that can be turned on/off with a specific prop. We could perhaps add this current guidance as a new Patterns section or something? (that can be a working name that we update in the future)

Really, we probably need _more_ guidance like this on our component pages. I just want to make sure we're not presenting it in a confusing way

Reply:
The amazing guidance and docs are all @hectoid - just the implementer : ) I've updated the names of the variants, Hector let me know what you think